File: Angular_test_project/ClientApp/src/app/models/SampleModel.ts
export interface ISampleModel {

  serialNumber: number;

  sample_Id: number;
  sampleType: string;
  customerSampleWeigth: number;
  actualSampleWeigth: number;
  weight_Unit: string;
  barNumber: string;
  returnSample: boolean;
  approved: boolean;

  sampleCertificateNotes: string;
  sampleLabNotes: string;
  sampleAdminNotes: string;
  customerSampleReference: string;

  expand: boolean;
}

export class SampleModel implements ISampleModel {
  serialNumber: number = 0;
  sample_Id: number = 0;
  sampleType: string = "";
  customerSampleWeigth: number = 0;
  actualSampleWeigth: number = 0;
  weight_Unit: string = "";
  barNumber: string = "";
  returnSample: boolean = false;
  approved: boolean = false;
  sampleCertificateNotes: string = "";
  sampleLabNotes: string = "";
  sampleAdminNotes: string = "";
  customerSampleReference: string = "";

  expand: boolean = false;
}

File: ModelCollection/SampleModel.cs
namespace ModelCollection
{
    public class SampleModel
    {
        public int SerialNumber { get; set; } //UNIQUE pair with FK

        public long Sample_Id { get; set; } //PK UNIQUE
        public string SampleType { get; set; }
        public decimal? CustomerSampleWeigth { get; set; }
        public decimal? ActualSampleWeigth { get; set; }
        public string Weight_Unit { get; set; }
        public string BarNumber { get; set; } //Nullable value is needed here
        public bool ReturnSample { get; set; } //is the sample being returned to the customer
        public bool Approved { get; set; } //has the sample been approved

        public string SampleCertificateNotes { get; set; }
        public string SampleLabNotes { get; set; }
        public string SampleAdminNotes { get; set; }
        public string CustomerSampleReference { get; set; } = "";
    }
}
